fpmsyncd/fpmlink.cpp

```diff
@@ -106,7 +106,7 @@ void FpmLink::readData()
     /* Check for complete messages */
     while (true)
     {
-        hdr = (fpm_msg_hdr_t *)(m_messageBuffer + start);
+        hdr = reinterpret_cast<fpm_msg_hdr_t *>(static_cast<void *>(m_messageBuffer + start));
```

Comment:
>static_cast<void *> [](start = 48, length = 19)

Could you remove static_cast and only use reinterpret_cast? Please also provide the compilation message before and after. We have no easy way to reproduce your problem.

Reply:
Original error log:
```
fpmlink.cpp:110:56: error: cast from 'char*' to 'fpm_msg_hdr_t* {aka fpm_msg_hdr_t_*}' increases required alignment of target type [-Werror=cast-align]
         hdr = (fpm_msg_hdr_t *)(m_messageBuffer + start);
```
Removing static cast error log:-
```
fpmlink.cpp:110:72: error: cast from 'char*' to 'fpm_msg_hdr_t* {aka fpm_msg_hdr_t_*}' increases required alignment of target type [-Werror=cast-align]
         hdr = reinterpret_cast<fpm_msg_hdr_t *>(m_messageBuffer + start);
```

Reply:
The original error does make sense, and original code need improvement to make it portable. casting is bad smell and double casting is even worse.

One possible solution is to refine original code to increase pointer by alignment. I notice [https://github.com/Azure/sonic-swss/blob/master/fpmsyncd/fpm/fpm.h#L116](https://github.com/Azure/sonic-swss/blob/master/fpmsyncd/fpm/fpm.h#L116) indicating alignment is always true.

---
In reply to: [310905179](https://github.com/Azure/sonic-swss/pull/1015#discussion_r310905179) [](ancestors = 310905179)

Reply:
I dont think the gcc warn about pack size of struct, it warns because the pointer to char can have lesser size compared to left-value  pointer reference.